tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/vectortypes/TestFloats.java

```diff
@@ -986,4 +1030,165 @@ public void testAllocationIssue() {
+        VectorFloat4 seq = new VectorFloat4(size);
+
+        for (int x = 0; x < b.getLength(); x++) {
+            a.set(x, new Float4(x, x, x, x));
+            b.set(x, new Int4(0, 1, 2, 3));
+        }
+
+        TaskGraph graph = new TaskGraph("s0") //
+                .transferToDevice(DataTransferMode.EVERY_EXECUTION, a, b) //
+                .task("t0", TestFloats::vectorComputationMixedTypes, a, b, output) //
+                .transferToHost(DataTransferMode.EVERY_EXECUTION, output);
+
+        ImmutableTaskGraph immutableTaskGraph = graph.snapshot();
+
+        vectorComputationMixedTypes(a, b, seq);
+
+        try (TornadoExecutionPlan executionPlan = new TornadoExecutionPlan(immutableTaskGraph)) {
+            executionPlan.execute();
+            for (int i = 0; i < size; i++) {
+                assertEquals(seq.get(i).getX(), output.get(i).getX(), DELTA);
+                assertEquals(seq.get(i).getY(), output.get(i).getY(), DELTA);
+                assertEquals(seq.get(i).getZ(), output.get(i).getZ(), DELTA);
+                assertEquals(seq.get(i).getW(), output.get(i).getW(), DELTA);
+            }
+        }
+    }
+
+    @Test
+    public void testVectorFloat8Int8() throws TornadoExecutionPlanException {
+        int size = 8192;
+        VectorFloat8 a = new VectorFloat8(size);
+        VectorInt8 b = new VectorInt8(size);
+        VectorFloat8 output = new VectorFloat8(size);
+        VectorFloat8 seq = new VectorFloat8(size);
+
+        for (int x = 0; x < b.getLength(); x++) {
+            a.set(x, new Float8(x, x, x, x, x, x, x, x));
+            b.set(x, new Int8(0, 1, 2, 3, 3, 2, 1, 0));
+        }
+
+        TaskGraph graph = new TaskGraph("s0") //
+                .transferToDevice(DataTransferMode.EVERY_EXECUTION, a, b) //
+                .task("t0", TestFloats::vectorComputationMixedTypes, a, b, output) //
+                .transferToHost(DataTransferMode.EVERY_EXECUTION, output);
+
+        ImmutableTaskGraph immutableTaskGraph = graph.snapshot();
+
+        vectorComputationMixedTypes(a, b, seq);
+
+        try (TornadoExecutionPlan executionPlan = new TornadoExecutionPlan(immutableTaskGraph)) {
+            executionPlan.execute();
+            for (int i = 0; i < size; i++) {
+                for (int k = 0; k < 8; k++) {
+                    assertEquals(seq.get(i).get(k), output.get(i).get(k), DELTA);
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testVectorFloat16Int16() throws TornadoExecutionPlanException {
+        int size = 8192;
+        VectorFloat16 a = new VectorFloat16(size);
+        VectorInt16 b = new VectorInt16(size);
+        VectorFloat16 output = new VectorFloat16(size);
+        VectorFloat16 seq = new VectorFloat16(size);
+
+        for (int x = 0; x < b.getLength(); x++) {
+            a.set(x, new Float16(x, x, x, x, x, x, x, x, x, x, x, x, x, x, x, x));
+            b.set(x, new Int16(0, 1, 2, 3, 3, 2, 1, 0, 0, 1, 2, 3, 3, 2, 1, 0));
```

Comment:
I have a suggestion for the initialization of the tested vectors. I would suggest to use different values for each field of the vectors, in order to ensure that indexing is correct and we do not have any accidental matches. What do you think?

Reply:
For this particular case it will have the same effect, because other tests already check with different values, but we can add random values instead.